### DISCUSSION TITLE
Now deployment file structure breaks paths for files

### DISCUSSION THREAD

**Discussion by v1rtl** [vercel/vercel#4242] [General]
I use file system to list all blog posts and then send them to API. Everything works flawlessly on a local computer but breaks in now deployment.

In Next.js docs it is written that I should use `process.cwd()` for accessing files, however, in now deployment, it fails, because it has a different file structure. Although, everything works fine with `next build && next start` and with `now dev`.

Here's the error I get on my deployment:

```
[GET] /api/posts
11:56:38:10
2020-05-02T08:56:38.126Z
ERROR
[Error: ENOENT: no such file or directory, scandir '/var/task/pages/blog']
{ 
  errno: -2,  code: 'ENOENT',  syscall: 'scandir',  path: '/var/task/pages/blog'
}
```

And here's what my [api endpoint](https://github.com/talentlessguy/v1rtl-2/blob/master/pages/api/posts.ts) looks like:

```ts
import { NextApiRequest, NextApiResponse } from 'next'
import { promises as fs } from 'fs'
import { join } from 'path'

const { readdir } = fs

export default async (req: NextApiRequest, res: NextApiResponse) => {
  const postFiles = await readdir(join(process.cwd(), 'pages/blog'))

  const postNames: string[] = postFiles.filter((page: string) => page !== 'index.tsx')

  const posts = []

  for (const post of postNames) {
    const mod = await import(`../blog/${post}`)

    posts.push({ ...mod, link: post.slice(0, post.indexOf('.')) })
  }

  res.status(200).json(posts)
}
```

The source file structure looks like this:

```
├── assets.d.ts
├── mdx.d.ts
├── next.config.js
├── next-env.d.ts
├── package.json
├── pages
│   ├── 404.tsx
│   ├── _app.tsx
│   ├── blog
│   │   ├── index.tsx
│   │   └── my-new-website.mdx
│   ├── _error.tsx
│   ├── index.tsx
│   └── list
│       └── [type]
│           └── [tags].tsx
├── pnpm-lock.yaml
├── README.md
└── tsconfig.json
```

The full source code is [here](https://github.com/talentlessguy/v1rtl-2), live deployment is [here](https://v1rtl.site/api/posts)

**Comment by mayankzalavadia** [comment]
@talentlessguy I also face the same issue. Did you find any solution?

My blog is working on local but it's giving me the same error on Vercel.

**Reply by v1rtl** [reply]
@mayankzalavadia not sure if a solution but I switched to next-mdx-remote
